go/mysql/mysql56_gtid_set.go

```diff
@@ -435,6 +435,99 @@ func (set Mysql56GTIDSet) SIDBlock() []byte {
 	return buf.Bytes()
 }
 
+func (set Mysql56GTIDSet) Difference(other GTIDSet) GTIDSet {
+	if other == nil {
+		return set
+	}
+	mydbOther, ok := other.(Mysql56GTIDSet)
+	if !ok {
```

Comment:
shouldn't this lead to an error instead of quietly throwing away `other`?

Reply:
My thought here is that we can't compare GTIDSets of differing flavors at at all, so the difference is purely the receiver.

That being said, because we can only do `Difference` for a `Mysql56GTIDSet` (and this is not an interface method), it probably makes more sense to accept `Mysql56GTIDSet` as the argument for `Difference` and let the type system strictly enforce this requirement.

Reply:
Done.